src/extension.ts

```diff
@@ -50,17 +63,17 @@ export function activate(context: vscode.ExtensionContext) {
   getTempDir(true /*recreate*/);
 
   // TBD: Can this be initialized when DHE is connected?
-  if (dheServerUrl) {
-    dheService = new DheService(dheServerUrl, outputChannel, dheWsUrl);
-    const fsService = new WebClientDataFsService(dheService.buildFsMap);
-    const webClientDataFs = new WebClientDataFsProvider(dheService, fsService);
-
-    context.subscriptions.push(
-      vscode.workspace.registerFileSystemProvider('dhfs', webClientDataFs, {
-        isCaseSensitive: true,
-      })
-    );
-  }
+  // if (dheServerUrl) {
+  //   dheService = new DheService(dheServerUrl, outputChannel, dheWsUrl);
+  //   const fsService = new WebClientDataFsService(dheService.buildFsMap);
+  //   const webClientDataFs = new WebClientDataFsProvider(dheService, fsService);
+
+  //   context.subscriptions.push(
+  //     vscode.workspace.registerFileSystemProvider('dhfs', webClientDataFs, {
```

Comment:
AFAIK, file system providers have to be registered up front. The actual client creation + connection should be lazy until the user actually opens a file or folder in the fs, but the registration has to be present so vscode recognizes the future request to a `dhfs://xxxx` resource.

Reply:
We could probably encapsulate the `DheService` creation inside of the FsProvider or something, but we'd want to cache the dhe client + session somewhere that is shared.